### ISSUE TITLE
Error in parallel: example_optimize_yaw_wind_rose_parallel.py

### ISSUE THREAD

**Issue by DeeDive** [NREL/floris#74]
**Bug description**
Hi, I tried to run example_optimize_yaw_wind_rose_parallel.py, and connection error is occurred.
After changing the wd & ws source from local file, the program can run, with following errors:

**To Reproduce**
In example_optimize_yaw_wind_rose_parallel.py, change download from API to False 
Run example_optimize_yaw_wind_rose_parallel.py

**Expected behavior**
Traceback (most recent call last):
  File "//floris/examples/optimization/scipy/example_optimize_yaw_wind_rose_parallel.py", line 127, in <module>
    df_base = yaw_opt.calc_baseline_power()
  File "/floris/floris/tools/optimization/scipy/optimization.py", line 1196, in calc_baseline_power
    for df_base_one in executor.map(self._calc_baseline_power_one_case,self.ws.values,self.wd.values):
  File "/mpi4py/futures/pool.py", line 207, in result_iterator
    yield futures.pop().result()
  File "/python3.8/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/env/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
TypeError: 'numpy.float64' object is not iterable

**Floris Version**
latest dev

**System Information (please complete the following information):** 
 - OS: Ubuntu 18.06

**Comment by NilsGaukroger** [comment]
I am also having the same issue.

File "DOW_model_parallel_Hamilton.py", line 78, in <module>
df_base = yaw_opt.calc_baseline_power()
File "/ddn/home/npcl74/floris/floris/tools/optimization/scipy/optimization.py", line 1196, in calc_baseline_power
for df_base_one in executor.map(self._calc_baseline_power_one_case,self.ws.values,self.wd.values):
File "/ddn/home/npcl74/.local/lib/python3.6/site-packages/mpi4py/futures/pool.py", line 207, in result_iterator
yield futures.pop().result()
File "/usr/local/Cluster-Apps/python/3.6.8/lib/python3.6/concurrent/futures/_base.py", line 425, in result
return self.__get_result()
File "/usr/local/Cluster-Apps/python/3.6.8/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
raise self._exception
TypeError: 'numpy.float64' object is not iterable

**System Information**

- OS: Linux - Hamilton Supercomputer

**Comment by bayc** [comment]
Hello @DeeDive , the connection issue is most likely to do with the server for the Wind Tool Kit. Sometimes it is down and unable to establish a connection. To use the locally stored wind rose information, you can change line 87 from `True` to `False`:
```
calculate_new_wind_rose = False
```
As for the `TypeError`, when you changed the wind speed and wind direction (assuming this was in `example_input.json`), did you supply both values as a list? Even if there is only one wind speed and/or one wind direction, they must be supplied as lists:
```
"wind_speed": [
  8.0
],
"wind_direction": [
  270.0
],
```
Let me know if you are still having trouble.

@NilsGaukroger , was this the correction you needed to make to get past this error?

**Comment by NilsGaukroger** [comment]
@bayc, sorry, should have made that clearer, I didn't have a connection error as I am running from a local pickled wind rose and not using Wind Tool Kit. My wind speeds and directions are already as lists, I think my error might have had something to do with not launching my python script with an "mpirun" command - as I am using an openmpi module.

**Comment by DeeDive** [comment]
@bayc Hi, I find that ws and wd in the config file have already been given as lists, and the error happened after serveral steps of computation.
Output:

Using default gauss deflection multipler of 1.2
Plotting the FLORIS flowfield...
Importing wind rose data...
Correcting negative Overhang:-7.5
Finding baseline and optimal wake steering power in FLORIS...
=====================================================
Calculating baseline power in parallel...
Number of wind conditions to calculate =  1872
=====================================================
Computing wind speed = 0.0 m/s, wind direction = 0.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 5.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 10.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 15.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 20.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 25.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 30.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 35.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 40.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 45.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 50.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 55.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 60.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 65.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 70.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 75.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 80.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 85.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 90.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 95.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 100.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 105.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 110.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 115.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 120.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 125.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 130.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 135.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 140.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 145.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 150.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 155.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 160.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 165.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 170.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 175.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 180.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 185.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 190.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 195.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 200.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 205.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 210.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 215.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 220.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 225.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 230.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 235.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 240.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 245.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 250.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 255.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 260.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 265.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 270.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 275.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 280.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 285.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 290.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 295.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 300.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 305.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 310.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 315.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 320.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 325.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 330.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 335.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 340.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 345.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 350.0 deg.
Computing wind speed = 0.0 m/s, wind direction = 355.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 0.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 5.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 10.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 15.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 20.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 25.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 30.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 35.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 40.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 45.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 50.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 55.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 60.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 65.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 70.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 75.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 80.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 85.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 90.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 95.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 100.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 105.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 110.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 115.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 120.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 125.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 130.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 135.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 140.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 145.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 150.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 155.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 160.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 165.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 170.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 175.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 180.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 185.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 190.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 195.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 200.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 205.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 210.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 215.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 220.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 225.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 230.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 235.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 240.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 245.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 250.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 255.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 260.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 265.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 270.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 275.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 280.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 285.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 290.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 295.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 300.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 305.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 310.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 315.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 320.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 325.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 330.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 335.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 340.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 345.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 350.0 deg.
Computing wind speed = 1.0 m/s, wind direction = 355.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 0.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 5.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 10.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 15.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 20.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 25.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 30.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 35.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 40.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 45.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 50.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 55.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 60.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 65.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 70.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 75.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 80.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 85.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 90.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 95.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 100.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 105.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 110.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 115.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 120.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 125.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 130.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 135.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 140.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 145.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 150.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 155.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 160.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 165.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 170.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 175.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 180.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 185.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 190.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 195.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 200.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 205.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 210.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 215.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 220.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 225.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 230.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 235.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 240.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 245.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 250.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 255.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 260.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 265.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 270.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 275.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 280.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 285.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 290.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 295.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 300.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 305.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 310.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 315.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 320.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 325.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 330.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 335.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 340.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 345.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 350.0 deg.
Computing wind speed = 2.0 m/s, wind direction = 355.0 deg.
Computing wind speed = 3.0 m/s, wind direction = 0.0 deg.
Computing wind speed = 3.0 m/s, wind direction = 5.0 deg.
Traceback (most recent call last):
  File "/home/student/wkspace/plantsim/plantsim/optimize_yaw_wind_rose_parallel_dev.py", line 128, in <module>
    df_base = yaw_opt.calc_baseline_power()
  File "/home/student/wkspace/plantsim/floris/floris/tools/optimization/scipy/optimization.py", line 1197, in calc_baseline_power
    for df_base_one in executor.map(self._calc_baseline_power_one_case,self.ws.values,self.wd.values):
  File "/home/student/.conda/envs/wind/lib/python3.7/site-packages/mpi4py/futures/pool.py", line 207, in result_iterator
    yield futures.pop().result()
  File "/home/student/.conda/envs/wind/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/home/student/.conda/envs/wind/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
TypeError: 'numpy.float64' object is not iterable

**Comment by bayc** [comment]
@DeeDive, @NilsGaukroger : Ok, thank you for the additional info. Currently, the latest develop branch for me runs this example without any issues. I believe the issue you are experiencing has been fixed in the last few days. Can you pull down the latest develop and see if you get the same issue?

Essentially, the `floris/tools/optimization/scipy/optimization.py` method `_calc_baseline_power_one_case` is receiving scalar values for wind speed and wind direction from the `MPIPoolExecutor()` for loop on  line 1194 of `optimization.py`. As such, when `_calc_baseline_power_one_case` calls `fi.reinitialize_flow_field`, scalar values for `ws` and `wd` are being passed when they should be lists. A fix has been implemented within `floris_interface.reinitialize_flow_field` to convert scalar values to lists. If pulling down the latest dev doesn't fix your issue (which it may not since it does seem that some calculations worked?), then can you please reply here with the output of `git log -1`? Then I will be able to dig into this further.

For example:
```
<user>:~/floris/examples/optimization/scipy$ git log -1
commit 383c8db1466d1f4bbfeb7f425578f8ae69938340 (HEAD -> feature/sowfa_tune, paul/feature/sowfa_tune)
Author: Paul <paul.fleming@nrel.gov>
Date:   Wed Mar 25 21:51:00 2020 -0600
```

**Comment by NilsGaukroger** [comment]
@bayc I've pulled the latest develop branch and now have this issue instead, the other issue may have been resolved though
```
  File "DOW_model_parallel_Hamilton.py", line 49, in <module>
    fi = wfct.floris_interface.FlorisInterface("DOW_input.json")
  File "/ddn/home/npcl74/floris/floris/tools/floris_interface.py", line 42, in __init__
    self.floris = Floris(input_file=input_file, input_dict=input_dict)
  File "/ddn/home/npcl74/floris/floris/simulation/floris.py", line 44, in __init__
    logging_dict = self.meta_dict["logging"]
KeyError: 'logging'
```
Also, my ```git log -1``` is:
```
commit 49d25ee09f07fe65a782c708b5d66c4497f56314
Merge: 390f2bb 1c25684
Author: Rafael M Mudafort <rafmudaf@gmail.com>
Date:   Wed Mar 18 12:53:13 2020 -0500

    Merge pull request #75 from paulf81/bugfix/changeturbine

    Bugfix/changeturbine
```

**Comment by DeeDive** [comment]
@bayc I have similar error. 

floris.simulation.wake_velocity.base_velocity_deficit INFO Using default option of calculating V and W velocity components (calculate_VW_velocities=False)
floris.simulation.wake_velocity.base_velocity_deficit INFO Using default option of applying added yaw-added recovery (use_yaw_added_recovery=True)
floris.simulation.wake_velocity.base_velocity_deficit INFO Using default option yaw_recovery_alpha: 0.03
floris.simulation.wake_velocity.base_velocity_deficit INFO Using default option eps_gain: 0.3
floris.simulation.wake_deflection.base_velocity_deflection INFO Using default option of applying gch-based secondary steering (use_secondary_steering=True)
floris.simulation.wake_deflection.base_velocity_deflection INFO Using default option eps_gain: 0.3
Traceback (most recent call last):
  File "/home/student/wkspace/plantsim/floris/examples/optimization/scipy/example_optimize_yaw_wind_rose_parallel.py", line 44, in <module>
    os.path.join(file_dir, '../../example_input.json')
  File "/home/student/wkspace/plantsim/floris/floris/tools/floris_interface.py", line 42, in __init__
    self.floris = Floris(input_file=input_file, input_dict=input_dict)
  File "/home/student/wkspace/plantsim/floris/floris/simulation/floris.py", line 51, in __init__
    self.farm = Farm(farm_dict, turbine, wake)
  File "/home/student/wkspace/plantsim/floris/floris/simulation/farm.py", line 93, in __init__
    specified_wind_height=properties["specified_wind_height"])
KeyError: 'specified_wind_height'

Process finished with exit code 1

Also, my `git log -1` is

commit 49d25ee09f07fe65a782c708b5d66c4497f56314 (origin/develop) 
Merge: 390f2bb 1c25684                                                           Author: Rafael M Mudafort                                    Date:   Wed Mar 18 12:53:13 2020 -0500                                                                                                                                Merge pull request #75 from paulf81/bugfix/changeturbine                                                                                                          Bugfix/changeturbine
--

**Comment by bayc** [comment]
@DeeDive @NilsGaukroger , alright, we recently pushed over a lot of changes to develop and think your issues should be solved with the latest version. I grab the latest develop, do:
```
git fetch --all
git checkout develop
```
That will pull down the latest version, and depending on when you last "installed" FLORIS, you may need to re-install to solve any `ModuleNotFoundError`. To do this, move out of the root directory and do:
```
pip install -e floris/
```
@DeeDive , I believe this should fix your error. 

@NilsGaukroger ,if you have developed other input files beyond the `example_input.json`, you may be missing some new fields that were recently added. @NilsGaukroger, I believe this to be your case as I can see from your code you are using an input file different than `example_input.json`. Look in `example_input.json` at line 5-13:
```
"logging": {
    "console": {
      "enable": false,
      "level": "INFO"
    },
    "file": {
      "enable": false,
      "level": "INFO"
    }
```
I believe this information is missing from your other input file.

**Comment by bayc** [comment]
@DeeDive Checking back on this issue, did the above resolve it?

**Comment by DeeDive** [comment]
Yes,@bayc it works now. Thanks a lot!

**Comment by bayc** [comment]
@DeeDive Good to hear! And we also just released v2.0 of FLORIS, moving the develop branch over to master, in case you are interested.
